src/test/ui/suggestions/field-access.fixed

```diff
@@ -0,0 +1,26 @@
+// run-rustfix
+#![allow(dead_code)]
+
+struct A {
+    b: B,
+}
+
+enum B {
+    Fst,
+    Snd,
+}
+
+fn main() {
+    let a = A { b: B::Fst };
+    if let B::Fst = a.b {};
+    //~^ ERROR mismatched types [E0308]
+    // note: you might have meant to use field `b` of type `B`
```

Comment:
The comment for the not is not being mechanically checked. For that, this would be 
```suggestion
    //~| HELP you might have meant to use field `b` of type `B`
```

the `|` is equivalent to "same line as the line above", so when you have three different checks instead of doing `^`, `^^` and `^^^`, you do `^` on the first and `|` on the rest.